packages/lesswrong/server/repos/UsersRepo.ts

```diff
@@ -346,12 +362,55 @@ export default class UsersRepo extends AbstractRepo<DbUser> {
           SELECT val
           FROM UNNEST(ARRAY_AGG(agreement_value)) AS val
           WHERE val != 0
-      ) AS agreement_values
-    FROM "CombinedVotes"
-    GROUP BY user_id, user_username, user_displayName
+      ) AS agreement_values,
+      uc.recently_active_matchmaking
+    FROM "CombinedVotes" as cv
+    LEFT JOIN "UserChecks" AS uc ON cv.user_id = uc._id
+    GROUP BY 
+      user_id, 
+      user_username, 
+      user_displayName,
+      uc.recently_active_matchmaking
     HAVING SUM(vote_power) > 1
     ORDER BY total_power DESC
     LIMIT $4;
-      `, [user._id, smallVotePower, bigVotePower, limit])
+      `, [user._id, smallVotePower, bigVotePower, limit, recencyLimitDays])
+  }
+
+
+  async getDialogueMatchedUsers(userId: string): Promise<DbUser[]> {
+    return this.any(`
+      SELECT DISTINCT(u.*)
+      FROM "DialogueChecks" other_users_checks
+      JOIN "DialogueChecks" current_user_checks
+      -- Join such that there must exist reciprocal checks
+      ON (
+        other_users_checks."targetUserId" = current_user_checks."userId"
+        AND current_user_checks."targetUserId" = other_users_checks."userId"
+        AND other_users_checks.checked IS TRUE
+        AND current_user_checks.checked IS TRUE
+      )
+      JOIN "Users" u
+      -- Given those, join for users who've created checks on you
+      ON (
+        other_users_checks."userId" = u._id
+        AND other_users_checks."targetUserId" = $1
+        AND current_user_checks."userId" = $1
+      )
+    `, [userId]);
+  }
+
+  async getActiveDialogueMatchSeekers(limit: number): Promise<DbUser[]> {
+    return this.manyOrNone(`
+      SELECT  
+        u.*,
+        MAX(dc."checkedAt") AS "mostRecentCheckedAt"
```

Comment:
You aren't using this in the actual results, just for the `ORDER BY`, right?  I expect it's getting stripped out of the results because you aren't using `.getRawDb()`, so if you needed it outside of the query it wouldn't be there.

Reply:
(If so, you shouldn't return it from the query, but just inline the `MAX(...)` expression into the `ORDER BY` clause directly.  We currently don't throw any errors when you return extra columns from queries that shouldn't be returning them, like this one, but plausibly we should - an issue that Oli and I debugged yesterday would've been caught by something like that.)

Reply:
(This doesn't need to be fixed before launch.)

Reply:
This was code you and Ben wrote yesterday

Reply:
and yes only used for ordering!